### DISCUSSION TITLE
Resteasy reactive: Multiple writers set at build time

### DISCUSSION THREAD

**Discussion by luca-bassoricci** [quarkusio/quarkus#26321] [Q&A] [answered]
Related to #26217

```@Path("{id}/json")
@GET
@Produces(MediaType.APPLICATION_JSON)
public RestResponse<File> getBatchJson(@PathParam("id") long batchId) throws IOException {
  Batch b = Batch.findById(batchId);
  if (b == null) {
    throw new NotFoundException();
  }
  File json = File.createTempFile("import", ".json");
  try (FileOutputStream outputStream = new FileOutputStream(json)) {
    outputStream.write(b.json);
  }
  return RestResponse.ResponseBuilder.ok(json, MediaType.APPLICATION_JSON)
    .header("Content-Disposition", "attachment; filename=\"import.json\"")
    .header("Content-Length", json.length())
 .build();
}
```

Score result:
1. org.jboss.resteasy.reactive.server.providers.serialisers.ServerFileBodyHandler@540e823e
2. io.quarkus.resteasy.reactive.jackson.runtime.serialisers.BasicServerJacksonMessageBodyWriter@6a0e81fd
3. org.jboss.resteasy.reactive.server.providers.serialisers.ServerStringMessageBodyHandler@860f2ac

which is the right approach to solve the problem? negotiate endpoint as generic `application/octet-stream` and specify `application/json` using `RestResponse.ResponseBuilder.ok(json, MediaType.APPLICATION_JSON)`?

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
The way things are now, I don't think you can get a perfect score. But I'll look into it as part of #26217
